src/Membrain/Memory.hs

```diff
@@ -32,6 +34,12 @@ newtype Memory (mem :: Nat) = Memory
     { unMemory :: Natural
     } deriving (Show, Eq)
 
+-- | Creates 'Memory' of unit 'mem' by given 'Natural' number.
+-- 'Memory's smart constructor.
```

Comment:
I think documentation is good enough for this function. It will become much clearer after we add examples :+1: